warehouse/query-core/src/main/java/datawave/query/tables/ssdeep/SSDeepScoringFunction.java

```diff
@@ -75,6 +85,8 @@ public SSDeepScoringFunction(SSDeepSimilarityQueryConfiguration config) {
 
         this.editDistanceScorer = new SSDeepHashEditDistanceScorer(maxRepeatedCharacters);
         this.ngramOverlapScorer = new SSDeepNGramOverlapScorer(config.getNGramSize(), maxRepeatedCharacters, config.getMinHashSize());
+
+        this.ngramCountMap = new HashMap<>();
```

Comment:
Not critical, but instead of owning/maintaining state here, consider moving the creation/descruction of the ngramCountMap map into the `SSDeepSimilarityQueryConfiguration` object where is other state is owned (e.g., config.getQueryMap())